libminifi/include/core/state/nodes/FlowInformation.h

```diff
@@ -110,32 +117,17 @@ class FlowMonitor : public StateMonitorNode {
     connection_store_.updateConnection(connection);
   }
 
- protected:
-  std::shared_ptr<state::response::FlowVersion> flow_version_;
-  ConnectionStore connection_store_;
-};
-
-/**
- * Justification and Purpose: Provides flow version Information
- */
-class FlowInformation : public FlowMonitor {
- public:
-  FlowInformation(std::string_view name, const utils::Identifier &uuid)
-      : FlowMonitor(name, uuid) {
-  }
-
-  explicit FlowInformation(std::string_view name)
-      : FlowMonitor(name) {
-  }
-
-  MINIFIAPI static constexpr const char* Description = "Metric node that defines the flow ID and flow URL deployed to this agent";
-
-  std::string getName() const override {
-    return "flowInfo";
+  void setProcessors(std::vector<core::Processor*> processors) {
```

Comment:
can the pointers be null? If yes, is it checked at the right places? If not, consider wrapping it in not_null.

Reply:
The processors come from the ProcessGroup's stored processor list, where it shouldn't be null, but unfortunately they are not enforced with not_null. Here the processor pointers are checked if they are null where they are used on line FlowInformation.cpp:67 and FlowInformation.cpp:102